### PR TITLE
feat: implement book page

### PR DESCRIPTION
**Pull request by SachkoMaxim**
- Implement api for catalog page;
- Implement book page and tests for it.

### CODE REVIEW

File: src/app/[bookId]/components/book-info.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+'use client'
+
+import { Star } from 'lucide-react'
```
Comment (tytianiuk) [inline src/app/[bookId]/components/book-info.tsx:3]:
```suggestion
import { Star } from 'lucide-react'
import Image from 'next/image'
```
import Image components from next

File: src/app/[bookId]/components/book-info.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+const BookInfo = ({ book }: BookInfoProps) => {
+  return (
+    <div>
+      <div className='md:flex'>
+        <div className='md:flex-shrink-0'>
+          <img
+            className='h-full min-w-72 w-full object-cover md:w-48'
+            src={'https://placehold.co/600x400'}
+            alt={book.name}
+          />
```
Comment (tytianiuk) [inline src/app/[bookId]/components/book-info.tsx]:
```suggestion
          <Image
            className='h-full min-w-72 w-full object-cover md:w-48'
            src={'https://placehold.co/600x400'}
            alt={book.name}
            width={100}
            height={100}
          />
```
replace img using Image

File: src/app/catalog/components/books-list.tsx [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
-  if (!books || books.length === 0) {
+  if (books.length === 0) {
     return <p className='text-lg font-medium'>Книг не знайдено</p>
   }
 
   return (
     <div className='md:col-span-3'>
       <div className='grid grid-cols-1 sm:grid-cols-2 2xl:grid-cols-3 gap-6'>
         {books.map((book) => (
```
Comment (tytianiuk) [inline src/app/catalog/components/books-list.tsx]:
```suggestion
  return (
    <div className='md:col-span-3'>
      <div className='grid grid-cols-1 sm:grid-cols-2 2xl:grid-cols-3 gap-6 h-[600px] overflow-y-auto'>
        {books.length === 0 ? (
          <div className='col-span-3 flex flex-col items-center justify-center h-[600px] text-center'>
            <BookX className='w-24 h-24 text-gray-400 mb-4' />
            <h2 className='text-2xl font-bold mb-2'>Книг не знайдено</h2>
            <span className='text-gray-600 mb-6 max-w-xl'>
              На жаль, за вашим запитом не знайдено жодної книги. Спробуйте
              змінити параметри пошуку або перегляньте наш повний каталог.
            </span>
          </div>
        ) : (
          books.map((book) => <BookCard book={book} key={book._id} />)
```

File: src/app/catalog/components/books-list.tsx [hunk 4]
```diff
@@ -1,34 +1,33 @@
 'use client'
 import { useSuspenseQuery } from '@tanstack/react-query'
-import { FC } from 'react'
```
Comment (tytianiuk) [inline src/app/catalog/components/books-list.tsx:4]:
```suggestion
import { BookX } from 'lucide-react'
```

File: src/app/catalog/components/books-list.tsx [hunk 5]
```diff
@@ -1,34 +1,33 @@
...
     return <p className='text-lg font-medium'>Книг не знайдено</p>
   }
 
   return (
     <div className='md:col-span-3'>
       <div className='grid grid-cols-1 sm:grid-cols-2 2xl:grid-cols-3 gap-6'>
         {books.map((book) => (
-          <BookCard book={book} key={book.id} />
+          <BookCard book={book} key={book._id} />
         ))}
```
Comment (tytianiuk) [inline src/app/catalog/components/books-list.tsx]:
```suggestion
        )}
```

File: src/app/page.tsx [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
 import { FC, Suspense } from 'react'
 
 import BooksFilters from '@/app/catalog/components/book-filters'
-import BooksList from '@/app/catalog/components/books-list'
 import { Filters } from '@/types/book'
 
+const DynamicBooksList = dynamic(
+  () => import('@/app/catalog/components/books-list'),
+  {
+    loading: () => <p>Завантаження...</p>,
```
Comment (tytianiuk) [inline src/app/page.tsx]:
```suggestion
   loading: () => (
      <div className='col-span-3 flex flex-col items-center justify-center h-[600px] text-center  '>
        <Loader2 className='col-span-3 self-center w-12 h-12 animate-spin text-gray-300' />
        <span className='text-2xl text-gray-300 font-bold mb-2'>
          Завантаження...
        </span>
      </div>
    ),
```
Here it need make design

File: src/app/page.tsx [hunk 7]
```diff
@@ -1,10 +1,17 @@
 import { Metadata } from 'next'
```
Comment (tytianiuk) [inline src/app/page.tsx:2]:
```suggestion
import { Loader2 } from 'lucide-react'
import { Metadata } from 'next'
```

File: src/app/[bookId]/components/book-comments-form.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  } = useForm({
+    defaultValues: { content: '' },
+  })
+
+  const allFields = watch()
+  const allFieldsFilled = allFields.content.trim() !== ''
+
+  const onSubmit = async (data: { content: string }) => {
+    if (!user) return
+    const comment = { userId: user._id, comment: data.content }
```
Comment (tytianiuk) [inline src/app/[bookId]/components/book-comments-form.tsx]:
```suggestion
    const comment = { userId: user!._id, comment: data.content }
```

there is no need to check, if there is no user, there will be no button with submit

File: src/app/[bookId]/components/book-review-form.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+  const allFields = watch()
+  const rating = watch('rating')
+
+  const allFieldsFilled = Object.values(allFields).every((value) => value)
+
+  const onSubmit = async (data: { content: string; rating: number }) => {
+    if (!user) return
+    const review = {
+      userId: user._id,
```
Comment (tytianiuk) [inline src/app/[bookId]/components/book-review-form.tsx]:
```suggestion
    const review = {
      userId: user!._id,
```
there is no need to check, if there is no user, there will be no button with submit

File: src/app/[bookId]/page.tsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
+'use client'
+
```
Comment (tytianiuk) [inline src/app/[bookId]/page.tsx:2]:
```suggestion

import { Loader2 } from 'lucide-react'
```

File: src/app/[bookId]/page.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+interface BookProps {
+  params: { bookId: string }
+}
+
+const Book = ({ params }: BookProps) => {
+  const [book, setBook] = useState<Book | null>(null)
+
+  const fetchBook = async () => {
+    const fetchedBook = await BooksAPI.getBookById(params.bookId)
+    setBook(fetchedBook)
```
Comment (tytianiuk) [inline src/app/[bookId]/page.tsx]:
```suggestion
    try {
      const fetchedBook = await BooksAPI.getBookById(params.bookId)
      setBook(fetchedBook)
    } catch {
      setBook(null)
    }
```

you need catch errors

File: src/app/[bookId]/page.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  useEffect(() => {
+    fetchBook()
+  }, [params.bookId])
+
+  if (!book) {
+    return <div>Книгу не знайдено</div>
```
Comment (tytianiuk) [inline src/app/[bookId]/page.tsx]:
```suggestion
    return (
      <div className='flex flex-col items-center justify-center h-[600px] text-center  '>
        <Loader2 className=' self-center w-12 h-12 animate-spin text-gray-300' />
        <span className='text-2xl text-gray-300 font-bold mb-2'>
          Завантаження...
        </span>
      </div>
    )
```
here it needs some design

**Review by tytianiuk — CHANGES_REQUESTED** [review]
I have done review, so you should see it

File: src/app/[bookId]/components/book-info.tsx (hunk 1 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/components/book-info.tsx:3]: Fixed it

File: src/app/[bookId]/components/book-info.tsx (hunk 2 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/components/book-info.tsx]: Fixed

File: src/app/catalog/components/books-list.tsx (hunk 3 above)
Comment (SachkoMaxim) [reply src/app/catalog/components/books-list.tsx]: Fixed

File: src/app/catalog/components/books-list.tsx (hunk 4 above)
Comment (SachkoMaxim) [reply src/app/catalog/components/books-list.tsx:4]: Fixed it

File: src/app/catalog/components/books-list.tsx (hunk 5 above)
Comment (SachkoMaxim) [reply src/app/catalog/components/books-list.tsx]: Fixed it

File: src/app/page.tsx (hunk 6 above)
Comment (SachkoMaxim) [reply src/app/page.tsx]: Fixed

File: src/app/[bookId]/components/book-comments-form.tsx (hunk 8 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/components/book-comments-form.tsx]: Fixed it

File: src/app/[bookId]/components/book-review-form.tsx (hunk 9 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/components/book-review-form.tsx]: Fixed

File: src/app/[bookId]/page.tsx (hunk 10 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/page.tsx:2]: Fixed it

File: src/app/[bookId]/page.tsx (hunk 11 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/page.tsx]: Fixed

File: src/app/[bookId]/page.tsx (hunk 12 above)
Comment (SachkoMaxim) [reply src/app/[bookId]/page.tsx]: Added

**Review by tytianiuk — APPROVED** [review]

**Pull request merged**